src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -426,6 +431,77 @@ def create(self, daemon_spec: CephadmDaemonSpec) -> str:
 
         return self.mgr._create_daemon(daemon_spec)
 
+    def create_realm_zonegroup_zone(self, spec: RGWSpec, rgw_id: str):
+        # check cluster health
+        ret, out, err = self.mgr.check_mon_command({
+            'prefix': 'health',
+            'format': 'json',
+        })
+        try:
+            j = json.loads(out)
+            self.mgr.log.info(j)
+        except Exception as e:
+            raise OrchestratorError('failed to parse health status')
+
+        if j['status'] != 'HEALTH_OK':
+                raise OrchestratorError('Health not ok, will try agin when health ok')
```

Comment:
Can you put this code into a new method? we probably want this make use of it elsewhere as well.

Reply:
i moved it to util let me know if thats not what you were thinking